app/controllers/password_resets_controller.rb

```diff
@@ -22,10 +22,12 @@ def create
   end
 
   def update
-    if @user.update_attributes(user_params)
-      log_in @user
+    update_password = UpdatePassword.call(user_params: user_params, user: @user)
+
+    if update_password.success?
+      log_in update_password.user
```

Comment:
Hey @npauzenga I think that method should be a `before_action`, you probably already want to have the user before updating it. That's normally how login is handled so that you don't have to put it in several places. What I like to do is create an `AuthenticatedController` that inherits from our `ApplicationController` and then add a `before_action` to it so that it logs the current user in. Every controller that requires authentication can just inherit from `AuthenticatedController` and if some action inside the controller does not require authentication, we can just add a skip filter.